server/tests/src/test/java/org/infinispan/server/security/authorization/AuthorizationGracefulShutdownIT.java

```diff
@@ -0,0 +1,100 @@
+package org.infinispan.server.security.authorization;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.infinispan.client.rest.RestResponseInfo.NO_CONTENT;
+import static org.infinispan.server.test.core.Common.sync;
+import static org.infinispan.server.test.core.TestSystemPropertyNames.INFINISPAN_TEST_SERVER_CONTAINER_VOLUME_REQUIRED;
+
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+import org.infinispan.client.rest.configuration.RestClientConfigurationBuilder;
+import org.infinispan.client.rest.configuration.RestClientConfigurationProperties;
+import org.infinispan.commons.test.Eventually;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.server.functional.ClusteredIT;
+import org.infinispan.server.test.api.TestUser;
+import org.infinispan.server.test.core.ContainerInfinispanServerDriver;
+import org.infinispan.server.test.core.ServerRunMode;
+import org.infinispan.server.test.junit5.InfinispanServerExtension;
+import org.infinispan.server.test.junit5.InfinispanServerExtensionBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class AuthorizationGracefulShutdownIT {
+
+   @RegisterExtension
+   public static InfinispanServerExtension SERVERS =
+         InfinispanServerExtensionBuilder.config("configuration/AuthorizationPropertiesTest.xml")
+               .runMode(ServerRunMode.CONTAINER)
+               .mavenArtifacts(ClusteredIT.mavenArtifacts())
+               .artifacts(ClusteredIT.artifacts())
+               .numServers(2)
+               .property(INFINISPAN_TEST_SERVER_CONTAINER_VOLUME_REQUIRED, "true")
+               .build();
+
+   @Test
+   public void testGracefulShutdownRestart() throws Exception {
+      TestUser user = TestUser.ADMIN;
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.clustering().cacheMode(CacheMode.DIST_SYNC).persistence().addSingleFileStore().segmented(false);
```

Comment:
Does the store matter for the test?

Reply:
Not really, no. It was a copy and paste from another test. It only verifies that the data is still there after restart.